### ISSUE TITLE
Use base2 log in entropy calculation

### ISSUE THREAD

**Issue by dodgejoel** [hlin117/mdlp-discretization#26]
The scipy entropy function called supports a base argument that can be used. Of course, the original paper's proof applies when entropy is calculated in base 2 not base e.

**Comment by hlin117** [comment]
I don't think this is necessary. The function should work just as well with `log_e` compared to `log_2`. Is there any advantage to using `log_2` instead of `log_e`?

I avoided `log_2` in the original code [here](https://github.com/hlin117/mdlp-discretization/blob/master/mdlp/_mdlp.pyx#L112) because it was unnecessary, and made the code perhaps harder to read.

**Comment by dodgejoel** [comment]
I think the theoretical results on optimality of the discretization obtained with this method specifically depend on the entropy calculations ,which enter into the decision to accept or reject a given cut-point, being calculated with `log_2`. At some point the authors even specifically call this out as a requirement for the validity of their results: "Note that this theorem requires that the entropy Ent(S) be defined using logarithms to the base 2" on page 1026.

I don't see how using a different base in the logarithm impacts code readability in a substantial way. When you say that the function works "just as well with `log_e` compared to `log_2`" what exactly do you mean?

**Comment by dodgejoel** [comment]
Perhaps you are claiming that the cut points produced will always be the same? Is it formally true that it does not matter which base is used in this algorithm?

**Comment by dodgejoel** [comment]
any response here?

**Comment by ChingChuan-Chen** [comment]
I checked this and I found out that it is different.
I use my implementation, the code is [here.](https://gist.github.com/ChingChuan-Chen/c887846ef14474ccf5158fa4d0f41fd3)

```python
# data generation
x = [139, 139, 139, 139, 1490, 1490, 1490, 32456, 32456, 33444, 33444, 33444, 35666, 35666, 35666, 35666] * 300
y = [1, 1, 1, 1, 2, 2, 2, 3, 4, 3, 3, 4, 3, 3, 4, 4] * 300

# use log
mdlpdisc = DiscretizationMDLP(x, y)
res = mdlpdisc.get_partition_points()
# res = [814.5, 16973.0, 32950.0, 34555.0]

# use log2 (replace log with log2 in Line 55 and Line 57)
mdlpdisc = DiscretizationMDLP(x, y)
res = mdlpdisc.get_partition_points()
# res = [814.5, 16973.0]
```

It will result in a different number of cut points when the sample size is not too large and not too small.
